homeassistant/components/here_travel_time/sensor.py

```diff
@@ -88,20 +99,67 @@
 NO_ROUTE_ERRORS = ["NGEO_ERROR_GRAPH_DISCONNECTED", "NGEO_ERROR_ROUTE_NO_END_POINT"]
 NO_ROUTE_ERROR_MESSAGE = "HERE could not find a route based on the input"
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+COORDINATE_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_APP_ID): cv.string,
-        vol.Required(CONF_APP_CODE): cv.string,
-        vol.Required(CONF_DESTINATION): cv.string,
-        vol.Required(CONF_ORIGIN): cv.string,
-        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
-        vol.Optional(CONF_MODE, default=TRAVEL_MODE_CAR): vol.In(TRAVEL_MODE),
-        vol.Optional(CONF_ROUTE_MODE, default=ROUTE_MODE_FASTEST): vol.In(ROUTE_MODE),
-        vol.Optional(CONF_TRAFFIC_MODE, default=False): cv.boolean,
-        vol.Optional(CONF_UNIT_SYSTEM): vol.In(UNITS),
+        vol.Inclusive(CONF_DESTINATION_LATITUDE, "coordinates"): cv.latitude,
+        vol.Inclusive(CONF_DESTINATION_LONGITUDE, "coordinates"): cv.longitude,
     }
 )
 
+PLATFORM_SCHEMA = vol.All(
+    cv.has_at_least_one_key(
+        CONF_DESTINATION_LATITUDE,
+        CONF_DESTINATION_SENSOR,
+        CONF_DESTINATION_DEVICE_TRACKER,
+        CONF_DESTINATION_PERSON,
+        CONF_DESTINATION_ZONE,
+        CONF_DESTINATION_ZONE_FRIENDLY_NAME,
+    ),
+    cv.has_at_least_one_key(
+        CONF_ORIGIN_LATITUDE,
+        CONF_ORIGIN_SENSOR,
+        CONF_ORIGIN_DEVICE_TRACKER,
+        CONF_ORIGIN_PERSON,
+        CONF_ORIGIN_ZONE,
+        CONF_ORIGIN_ZONE_FRIENDLY_NAME,
+    ),
+    PLATFORM_SCHEMA.extend(
+        {
+            vol.Required(CONF_APP_ID): cv.string,
+            vol.Required(CONF_APP_CODE): cv.string,
+            vol.Inclusive(
+                CONF_DESTINATION_LATITUDE, "destination_coordinates"
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_DESTINATION_LONGITUDE, "destination_coordinates"
+            ): cv.longitude,
+            vol.Exclusive(CONF_DESTINATION_LATITUDE, "destination"): cv.latitude,
+            vol.Exclusive(CONF_DESTINATION_SENSOR, "destination"): cv.entity_id,
+            vol.Exclusive(CONF_DESTINATION_DEVICE_TRACKER, "destination"): cv.entity_id,
+            vol.Exclusive(CONF_DESTINATION_PERSON, "destination"): cv.entity_id,
```

Comment:
Don't we handle sensor, device_tracker and person entities the same? If both the option value type is the same and the handling is the same, we just need one option for those.

Reply:
Indeed we do. We could use `origin_entitiy_id`.
What is still different is the zone friendly name. I copied it from waze/google but to be honest I don't understand why we need it in addition to the possibility to just defining the zone entity_id.

Reply:
@bah2830 you added the option to use the friendly name of a zone to `google_travel_time`, @Petro31 you added it to `waze_travel_time`. Are there use cases behind it or did you just implement it because it could be done?

Reply:
The feedback seems to be "some users requested it". @MartinHjelmare what way should we go?
1. Remove the zone friendly name resolving feature. Only allow `origin_latitude`, `origin_longitude` and `origin_entity_id`
2. Add it as `origin_zone_friendly_name`
3. Add  `origin_entity` which allows entity_ids and friendly names

I'd prefer option 1. because I cannot see a usecase which needs a friendly name instead of the entity_id itself.

Reply:
Option 1 sounds good. :+1:

We don't need to add all options in this first PR.

Reply:
Implemented